gammapy/datasets/simulate.py

```diff
@@ -26,30 +28,160 @@ class MapDatasetEventSampler:
     def __init__(self, random_state="random-seed"):
         self.random_state = get_random_state(random_state)
 
-    def _sample_coord_time(self, npred, temporal_model, gti):
-        data = npred.data[np.isfinite(npred.data)]
-        n_events = self.random_state.poisson(np.sum(data))
+    def _make_table(self, coords, time_ref):
+        """Create a table for sampled events.
 
-        coords = npred.sample_coord(n_events=n_events, random_state=self.random_state)
+        Parameters
+        ----------
+        coords : `~gammapy.maps.MapCoord`
+            Coordinates of the sampled events.
+        time_ref : `~astropy.time.Time`
+            reference time of the event list.
 
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table of the sampled events.
+        """
         table = Table()
         try:
             energy = coords["energy_true"]
         except KeyError:
             energy = coords["energy"]
 
+        table["TIME"] = u.Quantity(
+            ((coords["time"].mjd - time_ref.mjd) * u.day).to(u.s)
+        ).to("s")
         table["ENERGY_TRUE"] = energy
         table["RA_TRUE"] = coords.skycoord.icrs.ra.to("deg")
         table["DEC_TRUE"] = coords.skycoord.icrs.dec.to("deg")
 
+        return table
+
+    def _evaluate_timevar_source(self, dataset, evaluator, time_axis=None):
+        """Calculate Npred for a given `dataset.model` by evaluating
+        it in region geometry.
+
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Map dataset.
+        evaluator : `~gammapy.datasets.evaluators.MapEvaluator`
+            Map evaluator.
+        time_axis : `~gammapy.Maps.MapAxis`
+            Axis of the time.
+
+        Returns
+        -------
+        dataset.npred() : `~gammapy.dataset.MapDataset`
+            Npred map.
+        """
+        energy_axis = MapAxis.from_edges(
+            dataset.geoms["geom"].axes["energy"].edges, name="energy_true"
+        )
+
+        target = evaluator.model.spatial_model.position
+        on_region = PointSkyRegion(center=target)
+        region_geom = RegionGeom.create(on_region, axes=[energy_axis])
+
+        flux = evaluator.model.evaluate_geom(region_geom.to_wcs_geom(), gti=dataset.gti)
+        region_exposure = dataset.exposure.to_region_nd_map(region_geom.center_skydir)
+
+        npred = flux * region_exposure * dataset.geoms["geom"].bin_volume()
+        dataset.npred().data = npred.value
+
+        return dataset.npred()
```

Comment:
```suggestion
        return npred
```
I suppose this is what is intended here right?